homeassistant/components/http/cors.py

```diff
@@ -27,30 +27,36 @@ def setup_cors(app, origins):
         ) for host in origins
     })
 
-    def allow_cors(route, methods):
+    cors_added = set()
+
+    def _allow_cors(route, config=None):
         """Allow cors on a route."""
-        cors.add(route, {
-            '*': aiohttp_cors.ResourceOptions(
-                allow_headers=ALLOWED_CORS_HEADERS,
-                allow_methods=methods,
-            )
-        })
+        if hasattr(route, 'resource'):
+            path = route.resource
+        else:
+            path = route
+
+        path = path.canonical
```

Comment:
@balloob this line is giving me issues on my Windows 10 pc. It gives me the error:
```
AttributeError: 'PlainResource' object has no attribute 'canonical'
```
(See issue: https://github.com/home-assistant/home-assistant/issues/15687 )

Reply:
Clear your `deps` folder in config. You must be running an old version of `aiohttp` somehow. [PlainResource class](https://github.com/aio-libs/aiohttp/blob/master/aiohttp/web_urldispatcher.py#L305-L314)